### PR TITLE
Correctly use argument isFlatBottom in HoneycombVertexMeshGenerator constructor

### LINKED ISSUE #135: Correctly use argument isFlatBottom in HoneycombVertexMeshGenerator constructor
This issue corresponds to the old Chaste trac ticket [#3107](https://chaste.cs.ox.ac.uk/trac/ticket/3107).

To reiterate, the `HoneycombVertexMeshGenerator` constructor  accepts the argument `isFlatBottom` but ignores its value. This is in contrast to `CylindricalHoneycombVertexMeshGenerator`, where if `isFlatBottom` is true then the bottom boundary of the mesh is 'flattened'.

The task is therefore to correct the usage of `isFlatBottom` in the `HoneycombVertexMeshGenerator` constructor

### PR DESCRIPTION
**Pull request by georgiabeeton**
This pull request closes #135 by amending the `HoneycombVertexMeshGenerator` constructor to correctly use the argument `isFlatBottom`, with associated test `TestFlatBottomMesh()` in the test suite `TestHoneycombVertexMeshGenerator`.

### CODE REVIEW

File: mesh/src/vertex/HoneycombVertexMeshGenerator.cpp [hunk 1]
```diff
@@ -34,6 +34,7 @@ OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 */
 
 #include "HoneycombVertexMeshGenerator.hpp"
+#include "CheckpointArchiveTypes.hpp"
```
Comment (mirams) [inline mesh/src/vertex/HoneycombVertexMeshGenerator.cpp]: I think this include should only feature in things actually making archives, so generally tests, and the odd "archiver" class?

**Review by mirams — CHANGES_REQUESTED** [review]
Probably remove the `#include "CheckpointArchiveTypes.hpp"` from the class, and probably the test as well if it doesn't make archives.

File: mesh/test/vertex/TestHoneycombVertexMeshGenerator.hpp [hunk 2]
```diff
@@ -39,6 +39,7 @@ OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 #include <cxxtest/TestSuite.h>
 #include "HoneycombVertexMeshGenerator.hpp"
 #include "PetscSetupAndFinalize.hpp"
+#include "CheckpointArchiveTypes.hpp"
```
Comment (mirams) [inline mesh/test/vertex/TestHoneycombVertexMeshGenerator.hpp]: also this one, as I don't think archiving is being done in added code.

Comment (AlexFletcher) [reply mesh/test/vertex/TestHoneycombVertexMeshGenerator.hpp]: Resolved by commit 69c7ae2.

File: mesh/src/vertex/HoneycombVertexMeshGenerator.cpp (hunk 1 above)
Comment (AlexFletcher) [reply mesh/src/vertex/HoneycombVertexMeshGenerator.cpp]: Resolved by commit 69c7ae2.

**Review by mirams — APPROVED** [review]

**Pull request merged**